src/main/java/com/datalogics/pdf/security/LunaHsmManager.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2016 Datalogics Inc.
+ */
+
+package com.datalogics.pdf.security;
+
+import com.safenetinc.luna.LunaCryptokiException;
+import com.safenetinc.luna.LunaException;
+import com.safenetinc.luna.LunaSlotManager;
+import com.safenetinc.luna.provider.LunaProvider;
+
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.Security;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+
+/**
+ * This class allows for connecting to a Luna SA Hsm Device.
+ */
+public final class LunaHsmManager implements HsmManager {
+
+    private final LunaSlotManager slotManager;
+    private final KeyStore lunaKeyStore;
+
+    private boolean isLoggedIn;
+
+    public static final String KEYSTORE_TYPE = "Luna";
+    public static final String PROVIDER_NAME = "LunaProvider";
+
+    /**
+     * Default no-arg constructor.
+     *
+     * @throws IOException an I/O operation failed or was interrupted
+     */
+    protected LunaHsmManager() throws IOException {
+        super();
+        slotManager = LunaSlotManager.getInstance();
+        lunaKeyStore = getLunaKeyStore();
```

Comment:
This line doesn't work: you can't obtain the Luna key store until _after_ you are logged in.